utam-core/src/main/java/utam/core/framework/element/NavigationImpl.java

```diff
@@ -98,4 +105,59 @@ public void closeWindow(String targetUrl) {
   public void closeWindow() {
     this.driverAdapter.close();
   }
+
+  @Override
+  public int getWindowCount() {
+    return this.driverAdapter.getWindowHandles().size();
+  }
+
+  @Override
+  public Window currentWindow() {
+    return new WindowImpl(this.factory);
+  }
+
+  @Override
+  public void setupWaitForNewWindow() {
+    oldWindowCount = getWindowCount();
+    oldWindowHandles = new HashSet<>(this.driverAdapter.getWindowHandles());
+
+    this.setupForNewWindow = true;
+  }
+
+  @Override
+  public Window waitForNewWindow() {
+    if (!this.setupForNewWindow) {
+      throw new UtamError(ERR_SETUP_NOT_RUN);
+    }
+
+    // Wait for the window count to increase indicating a new window
+    this.driverAdapter.waitFor(() -> getWindowCount() > oldWindowCount, "wait for a new window to open", null);
```

Comment:
Change
`getWindowCount() > oldWindowCount`
to
`getWindowCount() > oldWindowHandles.size()`